### DISCUSSION TITLE
Where to find information about release planning

### DISCUSSION THREAD

**Discussion by seiferma** [eclipse-tycho/tycho#438] [Q&A] [answered]
Dear all,

I am really looking forward to the 2.6 release and I was wondering when it will be released. Do you have a page about release planning, on which I can look up such information? If not, you might want to think about creating such a page ;)

Best regards
Stephan

**Comment by laeubi** [comment]
Generally there are no planed release dates. Tycho is released whenever committers decide its time and/or useful and no critical actions taking place.

The 2.6.x release will most likely be released at end of January, you can help in this progress by [testing the current snapshot](https://github.com/eclipse/tycho/wiki#getting-tycho-snapshots) and give feedback if any problems or regressions occur for you.

**Reply by SchubP** [reply]
We are currently switching with oure huge codebase to J17 and are using the org.eclipse.tycho:tycho-compiler-jdt with the 2.6.0-SNAPSHOT for quite some time now (since [https://github.com/eclipse/tycho/commit/f1dada9c0c24cf7b74464e828556c6eb03fb14d8](https://github.com/eclipse/tycho/commit/f1dada9c0c24cf7b74464e828556c6eb03fb14d8) 9. Dec 2021) withouth any troubles. Looks really good to us. But we are desperately waiting for a release so that we can go in production (hopefully 1.1.2022).

**Reply by laeubi** [reply]
Just wondering as I assume your product will not be shipped with tycho why do you need a released version of tycho to go into production?

Given that we have at a lest one week to discover regressions, there is no preparation of the release yet and there is one feature missing on the roadmap I like to include first and the overall situation (holidays, chrismas, ...) its very unlikely to have a release this year.

**Reply by SchubP** [reply]
Hey, thx for the instant reply! :-) Well, of course we don't ship the tycho, but we build many products with the tycho and J17 and are rolling it out to all our developers at once with a big bang as J17+tycho+... require many changes wich prevent a step by step rollout/update and thus also makes it almost impossible for us to rollback or at least really expensive and thus we can only rollout stable releases to our development toolchain.

OK, given your answer I keep my fingers crossed for let's say the first week of January but also prepare a shift in our roadmap. To be fair we also assumed the tycho to be released with the eclipse 2021-12, but well, we "assumed", which implies a risk :-(

If there is anything we can test/do let us know. Unfortunatly we will also reduce/stop development over the next weeks :-/

**Reply by akurtakov** [reply]
Assumptions should be based on data too https://projects.eclipse.org/projects/technology.tycho/ . Tycho always releases multiple weeks after Eclipse to give time for integration of new features. Assuming your main goal is being able to produce Java 17 bytecode -you should be able to override ecj dependency of tycho-compiler-plugin 2.5.0 with the version from https://github.com/eclipse/tycho/commit/f1dada9c0c24cf7b74464e828556c6eb03fb14d8 as it's done in platform (https://git.eclipse.org/c/platform/eclipse.platform.releng.aggregator.git/tree/eclipse-platform-parent/pom.xml#n432) to test ecj milestones. That way you should be able to use released Tycho with released ECJ right now.
P.S. This should not be an excuse to not update to latest Tycho version when available as the project only supports latest release.

**Reply by laeubi** [reply]
> Unfortunatly we will also reduce/stop development over the next weeks

I think as with most companies/users those days, that's why I think its not wise to rush here as its unlikely that many people testing the snapshot before the release so to be realistic we have to extend the "grace-period" at least to the second week of January even if we start the release process next week.

> If there is anything we can test/do let us know

In general, if Tycho is key technology for your company, you can help with the following things:

1. Test snapshots of Tycho early and often so that regressions are found early before we start the release process, this gives faster release and maybe even more often releases if we are certain that snapshots are production-ready.
2. Consider donate some developer time to improve code (e.g. fixing bugs), enhancing documentation, help with review of open PRs, [answer questions on the discussions](https://github.com/eclipse/tycho/discussions), or [providing integration test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) so we have better feature coverage and thus faster releases.
3. You can [sponsor me](https://github.com/sponsors/laeubi) so I can assign more development-time-frames to Tycho.

**Reply by SchubP** [reply]
Of course, just to be sure, I didn't want to point any fingers or so, if, I would point at myself for "assuming" ;-)
Yeah, good point, maybe we will go with that workaround for a first stage ...

Yes, that's why we are decided using the snapshot, to test it, but so far we didn't run into any problem. The other 2 points are valid and are on oure mind and if we find anything we usually fix it (until now only in other projects like maven-plugins but not in tycho so far) and the sponsoring should be discussed in our next meeting.

**Comment by laeubi** [comment]
@akurtakov @mickaelistria What do you think if we branch from commit 1fe29d7ddbb31213230bb3bc75a90a8268467d9e a 2.6.0 release right now and delay my recent changes to 2.7.x?

**Reply by akurtakov** [reply]
Sounds good to me.

**Reply by laeubi** [reply]
Alright, I'll start to prepare the release then see https://github.com/eclipse/tycho/issues/490

**Comment by laeubi** [comment, marked as answer]
By the way, the official release planning site is here: https://projects.eclipse.org/projects/technology.tycho

For the 2.6.0 release please see https://github.com/eclipse/tycho/discussions/295
If you have tested the SNAPSHOT and all is fine, please leave a message there, if you encounter any issue please add a note as well and open a bug report!
